src/services/goalTemplates.ts

```diff
@@ -0,0 +1,354 @@
+      ],
+      where: { goalTemplateId },
+      order: [['ordinal', 'asc']],
+      raw: true,
+    }),
+    GoalFieldResponseModel.findAll({
+      attributes: [
+        ['goalTemplateFieldPromptId', 'promptId'],
+        'response',
+        [
+          sequelize.fn('ARRAY_AGG', sequelize.fn('DISTINCT', sequelize.col('"GoalFieldResponse"."goalId"'))),
+          'goalIds',
+        ],
+      ],
+      where: { goalId: goalIds },
+      include: [{
+        model: GoalTemplateFieldPromptModel,
+        as: 'prompt',
+        required: true,
+        attributes: [],
+        include: [{
+          model: GoalTemplateModel,
+          as: 'goalTemplate',
+          required: true,
+          attributes: [],
+          where: { id: goalTemplateId },
+        }],
+      }],
+      group: [
+        '"GoalFieldResponse"."goalTemplateFieldPromptId"',
+        '"GoalFieldResponse"."response"',
+      ],
+      raw: true,
+    }),
+  ]);
+
+  // restructure the collected data into one object with all responses for the passed goalIds if
+  // any exists
+  const restructuredPrompts = responses.reduce(
+    (
+      promptsWithResponses: FieldPrompts[],
+      response: PromptResponse,
+    ) => {
+      const exists = promptsWithResponses
+        .find((pwr) => pwr.promptId === response.promptId);
+      if (exists) {
+        exists.responses.push({
+          response: response.response,
+          goalIds: response.goalIds,
+        });
+      }
+      return promptsWithResponses;
+    },
+    prompts,
+  );
+  return restructuredPrompts;
+}
+
+/**
+
+Sets the field prompt response for a curated template for a list of goals.
+
+@param {number[]} goalIds - The IDs of the goals to update.
+@param {number} promptId - The ID of the field prompt to update.
+@param {string[] | null} response - The response to set for the field prompt.
+@returns {Promise} A Promise that resolves when the update is complete.
+*/
+export async function setFieldPromptForCuratedTemplate(
+  goalIds: number[],
+  promptId: number,
+  response: string[] | null,
+) {
+  // Retrieve the current responses and prompt requirements for the given goals and prompt ID.
+  const [currentResponses, promptRequirements] = await Promise.all([
+    GoalModel.findAll({
+      attributes: [
+        [sequelize.col('Goal.id'), 'goalId'],
+        [sequelize.col('prompts.id'), 'promptId'],
+        [sequelize.col('prompts.ordinal'), 'ordinal'],
+        [sequelize.col('responses.response'), 'response'],
+        [
+          sequelize.literal(`COALESCE("GoalFieldResponseModel"."response",array[]::text[]) != ARRAY[${response
+            ?.map((r) => sequelize
+              .getQueryInterface()
+              .queryGenerator
+              .escape(r))
+            .join(',')}]::text[]`),
+          'isChanged',
+        ],
+      ],
+      where: {
+        id: goalIds,
+        [Op.or]: [
+          { '$"prompts"."id"$': '$"responses"."goalTemplateFieldPromptId"$' },
```

Comment:
This line generates a syntax error

Here is the SQL currently being generated:
```sql
SELECT "Goal"."id" AS "_0",
	"prompts"."id" AS "_1",
	"prompts"."ordinal" AS "_2",
	"responses"."response" AS "_3",
	"GoalFieldResponseModel"."response" != ARRAY['Community Options'] AS "_4",
	"prompts->GoalFieldResponse"."id" AS "_5",
	"prompts->GoalFieldResponse"."goalId" AS "_6",
	"prompts->GoalFieldResponse"."goalTemplateFieldPromptId" AS "_7",
	"prompts->GoalFieldResponse"."response" AS "_8",
	"prompts->GoalFieldResponse"."onAR" AS "_9",
	"prompts->GoalFieldResponse"."onApprovedAR" AS "_10",
	"prompts->GoalFieldResponse"."createdAt" AS "_11",
	"prompts->GoalFieldResponse"."updatedAt" AS "_12"
FROM "Goals" AS "Goal"
LEFT OUTER JOIN "GoalFieldResponses" AS "responses" ON "Goal"."id" = "responses"."goalId"
AND "responses"."goalTemplateFieldPromptId" = 1
INNER JOIN ("GoalFieldResponses" AS "prompts->GoalFieldResponse"
INNER JOIN "GoalTemplateFieldPrompts" AS "prompts" ON "prompts"."id" = "prompts->GoalFieldResponse"."goalTemplateFieldPromptId") ON "Goal"."id" = "prompts->GoalFieldResponse"."goalId"
AND "prompts"."id" = 1
WHERE ("prompts"."id" = '$"fieldResponses"."goalTemplateFieldPromptId"$'
	OR "prompts"."goalTemplateFieldPromptId" IS NULL)
	AND "Goal"."id" IN (37499)
```